src/Sylius/Bundle/ApiBundle/spec/Security/OrderAdjustmentsVoterSpec.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Security;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ShopUser;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+
+final class OrderAdjustmentsVoterSpec extends ObjectBehavior
+{
+    function it_does_not_support_other_attributes(): void
+    {
+        $this->supportsAttribute('OTHER_ATTRIBUTE')->shouldReturn(false);
+    }
+
+    function it_votes_granted_when_collection_is_empty(TokenInterface $token): void
+    {
+        $collection = new ArrayCollection();
+
+        $this->vote($token, $collection, ['SYLIUS_ORDER_ADJUSTMENT'])->shouldReturn(VoterInterface::ACCESS_GRANTED);
+    }
+
+    function it_votes_granted_when_order_user_matches_token_user(
+        TokenInterface $token,
+        Collection $collection,
+        AdjustmentInterface $adjustment,
+        OrderInterface $order,
+    ): void {
+        $user = new ShopUser();
+
+        $token->getUser()->willReturn($user);
+        $collection->isEmpty()->willReturn(false);
+        $collection->first()->willReturn($adjustment);
+        $adjustment->getOrder()->willReturn($order);
+        $order->getUser()->willReturn($user);
+        $order->isCreatedByGuest()->willReturn(false);
+
+        $this->vote($token, $collection, ['SYLIUS_ORDER_ADJUSTMENT'])->shouldReturn(VoterInterface::ACCESS_GRANTED);
+    }
+
+    function it_votes_denied_when_order_user_does_not_match_token_user(
```

Comment:
Seems we're missing cases of:
1. token has a user; order does not
2. order has a user; token does not